beacon-chain/rpc/eth/blob/handlers.go

```diff
@@ -0,0 +1,174 @@
+				return
+			}
+		} else {
+			slot, err := strconv.ParseUint(blockId, 10, 64)
+			if err != nil {
+				errJson := &network.DefaultErrorJson{
+					Message: lookup.NewBlockIdParseError(err).Error(),
+					Code:    http.StatusBadRequest,
+				}
+				network.WriteError(w, errJson)
+				return
+			}
+			denebStart, err := slots.EpochStart(params.BeaconConfig().DenebForkEpoch)
+			if err != nil {
+				errJson := &network.DefaultErrorJson{
+					Message: errors.Wrap(err, "could not calculate Deneb start slot").Error(),
+					Code:    http.StatusInternalServerError,
+				}
+				network.WriteError(w, errJson)
+				return
+			}
+			if primitives.Slot(slot) < denebStart {
+				errJson := &network.DefaultErrorJson{
+					Message: "blobs are not supported before Deneb fork",
+					Code:    http.StatusBadRequest,
+				}
+				network.WriteError(w, errJson)
+				return
+			}
+			sidecars, err = s.BeaconDB.BlobSidecarsBySlot(r.Context(), primitives.Slot(slot), indices...)
+			if err != nil {
+				errJson := &network.DefaultErrorJson{
+					Message: errors.Wrapf(err, "could not retrieve blobs for slot %d", slot).Error(),
+					Code:    http.StatusInternalServerError,
+				}
+				network.WriteError(w, errJson)
+				return
+			}
+			network.WriteJson(w, buildSidecardsResponse(sidecars))
+			return
+		}
+	}
+
+	var err error
+	sidecars, err = s.BeaconDB.BlobSidecarsByRoot(r.Context(), bytesutil.ToBytes32(root), indices...)
+	if err != nil {
+		errJson := &network.DefaultErrorJson{
+			Message: errors.Wrapf(err, "could not retrieve blobs for root %#x", root).Error(),
+			Code:    http.StatusInternalServerError,
+		}
+		network.WriteError(w, errJson)
+		return
+	}
+	network.WriteJson(w, buildSidecardsResponse(sidecars))
+}
+
+// parseIndices filters out invalid and duplicate blob indices
+func parseIndices(url *url.URL) []uint64 {
+	query := url.Query()
+	helpers.NormalizeQueryValues(query)
+	rawIndices := query["indices"]
+	indices := make([]uint64, 0, field_params.MaxBlobsPerBlock)
+loop:
+	for _, raw := range rawIndices {
+		ix, err := strconv.ParseUint(raw, 10, 64)
+		if err != nil {
+			continue
+		}
+		for i := range indices {
+			if ix == indices[i] || ix >= field_params.MaxBlobsPerBlock {
+				continue loop
+			}
+		}
+		indices = append(indices, ix)
+	}
+	return indices
+}
+
+func buildSidecardsResponse(sidecars []*eth.BlobSidecar) *SidecarsResponse {
+	resp := &SidecarsResponse{Data: make([]*Sidecar, len(sidecars))}
+	for i, sc := range sidecars {
+		resp.Data[i] = &Sidecar{
+			BlockRoot:       hexutil.Encode(sc.BlockRoot),
+			Index:           strconv.FormatUint(sc.Index, 10),
+			Slot:            strconv.FormatUint(uint64(sc.Slot), 10),
+			BlockParentRoot: hexutil.Encode(sc.BlockParentRoot),
+			ProposerIndex:   strconv.FormatUint(uint64(sc.ProposerIndex), 10),
+			Blob:            hexutil.Encode(sc.Blob),
+			KZGCommitment:   hexutil.Encode(sc.KzgCommitment),
+			KZGProof:        hexutil.Encode(sc.KzgProof),
+		}
+	}
```

Comment:
I dont know what the best practice is here and whether we need to copy. Lgtm otherwise

Reply:
lets keep an eye out on it in testing. we have a few more PRs that will need to touch this too will keep an eyeout on ssz